misc-traits.js

```diff
@@ -292,4 +292,66 @@ class OptimalComparisonTrait extends Trait {
 
         return `Optimal Comparison: ${stg.optComp}%\n`;
     }
+}
+
+class UserSpecificMessageTrait extends Trait {
+    name = "secrets"
+
+    onStart(state) {
+        const stg = super.onStart(state);
+        stg.uuid = withDef(this.stg(state.stats).uuid, Math.floor(Math.random() * 1000000));
+        this.onReload();
+    }
+
+    onSave(state) {
+        this.stg(state.stats).uuid = this.stg(state.data).uuid;
+    }
+    
+    onPreShare(state) {
+        if (this.stg(state.stats).uuid) {
+            let bin = this.stg(state.stats).uuid.toString(2);
+            console.log(this.stg(state.stats).uuid);
+            let invis = bin.replace(/0/g, '\u200B').replace(/1/g, '\u200D');
+            return invis;
+        }
+        return "";
+    }
+
+    onReload(state) {
+        const msg = this.getSecret(state);
+        msg.then(data => {
+            if (data) {
+                console.log(data);
+                const box = docMake("div", ["overlay-box", "normal-text"]);
+                docMake("div", ["header-text", "gift-title"], box, e => {
+                    e.innerText = data;
+                    e.addEventListener("click", _e => {
```

Comment:
Do we just let them figure out they have to click it to make it go away?

Reply:
yes

Reply:
it isn't that unintuitive